app/services/listing_index_service/search/sphinx_adapter.rb

```diff
@@ -19,6 +19,12 @@ def search(community_id:, search:, includes:)
         Result::Error.new(ArgumentError.new("Both DB query and search engine would be needed to fulfill the search"))
       end
 
+      # rename listing_shape_ids to singular so that Sphinx understands it
+      if search[:listing_shape_ids] && search[:listing_shape_id].nil?
+        search[:listing_shape_id] = search[:listing_shape_ids]
+        search.delete(:listing_shape_ids)
+      end
+
```

Comment:
@rap1ds Thanks for good advice! Didn't know about us having the HashUtils.
And well spotted with needs_search?
I'll deploy to production when tests green